### ISSUE TITLE
Message Preview is not working - Certificate issue

### ISSUE THREAD

**Issue by JasGo00** [OfficeDev/microsoft-teams-apps-company-communicator#1553]
after renewal of the secret for the Author app and Graph app the preview message does not work with following error:

![image](https://github.com/user-attachments/assets/3b49cf58-db66-4b21-b932-f22aef143379)

I have updated the new secret to all necessary places withing function apps and the app service.

**Comment by v-jaygupta** [comment]
Hello @JasGo00 ,

Thanks for reaching out to us!
Could you please confrim if this is a new deployment or old? If its a new deployment please verify the secret value is copied and updated and not the client secret id for the app registrations. You need to follow the below steps to generate new client secrets and update at respective places - 


- Go to Azure Portal
- Go to App registration
- Create a new secret value for app registration and grab secret value. Here is the reference.
![image](https://github.com/user-attachments/assets/28c94134-1778-4354-80e0-76f0b7d2a56a)
- Then go to resource group. Select key vault from list of resources.
- Click on secrets under object section.
![image](https://github.com/user-attachments/assets/5e31b71c-7b05-405c-8b63-83da3d2fec76)
 - If the secrets are not visible, Go to Access Policy--> Click on Create
 ![image](https://github.com/user-attachments/assets/36fe126f-c87e-4872-ae36-efe715f73762)
- Select **Key, secret and certificate management**
![image](https://github.com/user-attachments/assets/760a91de-8d14-4847-a7d2-3da8c4f9e7bd)
- Give the Id through which one can view the secrets
![image](https://github.com/user-attachments/assets/a85f2ad2-3c26-44a9-9a1e-26c24d7f7552)
- 3rd step is optional
- Go for Review+Create
- Then go to Secrets. Open author/user/graph app password
- Click new version
![image](https://github.com/user-attachments/assets/870351f1-52bf-4297-a059-8c86ed644391)
- Grab the new secret identifier
![image](https://github.com/user-attachments/assets/6af0bd19-b84a-4faf-84e1-af4473d92d50)
- Paste it in the configuration sections of app service, 3-function apps. Here is the reference of updating the author app password in app service configuration.
![image](https://github.com/user-attachments/assets/f1d49102-affe-4447-8aa1-d2b388e00576)
- Follow the same procedure for the user app password and graph app password updating.
- For graph app, please dont forget to update the password in AzureAd Client Secret as well. here is the reference
![image](https://github.com/user-attachments/assets/bbb34da5-911d-40c1-b4e3-6a5dbf705be0)

**Comment by JasGo00** [comment]
Hello
this is an old deployment... I just renewed the certificate for Graph and Author apps...I have done all these steps and checked it 3 times, but the Preview is not working. this is the only thing that is not working

**Comment by peddivyshnavi** [comment]
Hello @JasGo00 

Please let us know if you are still facing this issue and require assistance and we will be happy to discuss. If no further action is needed at the moment and no response is received, we will close out the issue in 2 business days.
You may also always open a new issue or respond here if something does come back up.

Regards,

**Comment by JasGo00** [comment]
Yes we still have the problem and need assistance with this issue. We can send messages as I stated in my first post, but the preview is not working

**Comment by peddivyshnavi** [comment]
Hi @JasGo00 
Could you please check whether you have updated author app id in author's manifest file 
reference:
![image](https://github.com/user-attachments/assets/52195790-adf9-48b8-907f-95055279f9e0)

could you please share email id so that we can start discussing the issue internally

**Comment by JasGo00** [comment]
OK, Thanks looking forward to the resolution

**Comment by JasGo00** [comment]
Hello.
I am not sure why do I need to update the app ID? That did not change.

However the key and the secret have been updated in the Manifest file

***@***.***

***@***.***



Best regards

Jasmin Gokovic

BD/WPA-UCS1 / MS Teams/Skype
Mobile +1(248)310-2978 | ***@***.******@***.***>

**Comment by peddivyshnavi** [comment]
we have connected internally for call regarding Message Preview issue

- upon investigating we got to know there is an issue with author and graph secrets. we have created new secrets and updated. after update we see issue got resolved. now we see preview functionality is working as expected.

- we see issue of 16,000 unreceived messages, and we have mentioned to go with manual query which will elimates users which doesn't have CC will be removed from group.

- we see there is issue in exporting detailed results we have cleared out the table for export data to attempt a fresh report 

Could you please let us know if you are still facing this issue with preview message

**Comment by JasGo00** [comment]
The issue with the preview has been resolved.... However I am still waiting on the query to be able to delete the stale User Data... can you please send that to me as soon as possible.
Thanks!

Best regards

Jasmin Gokovic

BD/WPA-UCS1 / MS Teams/Skype
Mobile +1(248)310-2978 | ***@***.******@***.***>

**Comment by JasGo00** [comment]
I was able to use this query with extracted user IDs to delete almost 17,000 lines in the User Data table. I need to wait for next company wide message to see the result of this change

**Comment by peddivyshnavi** [comment]
Hi @JasGo00 
Marking this ticket as closed as we see issue with preview message got resolved 
Could you please open a new ticket regarding another issue so that it will be easy to track the issue.
Thanks
